release-calendar/src/server/app.ts

```diff
@@ -68,6 +74,40 @@ async function main(): Promise<void> {
     app.get('/', async (req, res) => {
       res.sendFile(path.join(DIST_DIR, 'index.html'));
     });
+
+    app.post('/insert/', async (req, res) => {
+      // authorization
+      const authorization = req.header('authorization');
+      const auth = authorization.split('Basic ')[1];
+      if (auth != process.env.BASIC) {
+        return res.status(401).json('Request is not authorized');
+      }
+
+      // parse ctime string into date
+      const {releases, promotions} = req.body;
+      promotions.forEach((p: {time: string; date: Date}) => {
+        p.date = moment(p.time, _CTIME_FORMAT).toDate();
+      });
+
+      const errors = {
+        createReleases: {},
+        createPromotions: {},
```

Comment:
Does this have to be initialized, or could the `catch` handlers below add properties to an otherwise empty object? `errors.createReleases` and `errors.createPromotions` below will always be truthy (`Boolean({}) === true`)

Reply:
Good catch. I turned errors into an array.